bin/run-script

```diff
@@ -0,0 +1,52 @@
+#! /usr/bin/env bash
+
+set -euo pipefail
+
+#### Detect Toolkit Project Root ####
+# if realpath is not available, create a semi-equivalent function
+command -v realpath >/dev/null 2>&1 || realpath() {
+  [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
+SCRIPT_PATH="$(realpath "${BASH_SOURCE[0]}")"
+SCRIPT_DIR="$(dirname "$SCRIPT_PATH")"
+TOOLKIT_ROOT="$(realpath "$SCRIPT_DIR/..")"
+if [[ ! -d "$TOOLKIT_ROOT/bin" ]] || [[ ! -d "$TOOLKIT_ROOT/config" ]]; then
+  echo "ERROR: could not find root of overleaf-toolkit project (inferred project root as '$TOOLKIT_ROOT')"
+  exit 1
+fi
+
+
+function usage() {
+  echo "Usage:
+    bin/run-script [SCRIPT...] [SCRIPT_ARGS..]
+    bin/run-script [OPTIONS]
+
+Options:
+    help              prints this help
+    ls                prints a list of all available scripts
+
+Examples:
+    bin/run-script create_project --user-id=649c3f45711ad101a13de737
+    bin/run-script ls"
+}
+
+
+function __main__() {
+  if [[ "${1:-null}" == "null" ]] \
+    || [[ "${1:-null}" == "help" ]] \
+    || [[ "${1:-null}" == "--help" ]] ; then
+    usage && exit
+  fi
+
+  if [[ "${1:-null}" == "ls" ]] || [[ "${1:-null}" == "--ls" ]]; then
+    # only prints files under services/web/scripts, no recursive lookup
+    local cmd='ls -p /overleaf/services/web/scripts | grep -v /'
+    exec "$TOOLKIT_ROOT/bin/docker-compose" exec sharelatex bash -c "$cmd"
+    exit
+  fi
+
+  local run_cmd="cd /overleaf/services/web; node scripts/$@"
+  exec "$TOOLKIT_ROOT/bin/docker-compose" exec sharelatex bash -c "$run_cmd"
```

Comment:
Some scripts load settings/talk to other services, which depend on the `/etc/container_environment.sh` / `/etc/overleaf/env.sh` env files being sourced.

Reply:
With the other comment in mind, we might need to omit the `scripts/` prefix here so that customers can invoke:
```
bin/run-script scripts/create_project
bin/run-script modules/server-ce-scripts/scripts/change-compile-timeout
```

Reply:
> With the other comment in mind, we might need to omit the scripts/ prefix here so that customers can invoke: 

That's done in https://github.com/overleaf/toolkit/commit/c377c94156aadd5f6ed309791be717d263c7f47c

Reply:
Right, with `node $@` it's currently not possible to set env-vars, but that might be necessary for some scripts.

WDYT about checking whether the first argument contains a `=` and if so run the bare `$@` (i.e. `bin/run-script FOO=bar node scripts/foo`, otherwise run `node $@`.

We can source the two files before running `$@`, to allow overriding any hosts etc.

Reply:
Pushed https://github.com/overleaf/toolkit/pull/277/commits/d4e100267499e4d1a0a81a509790ffd57ace0143 sourcing `container_environment.sh` and `env.sh` (in either `sharelatex/` or `overleaf/` location)

I added an `-e` option that can contain a space-separated list of environment variables to be included in the command. I think I prefer this option rather than a loose argument regexp parsing. Let me know what you think.

Reply:
Nice!